0.23.0 fails to compile under gcc 10.1

```
$ spack install flux-core@0.23.0 %gcc@10.1.0
<snip>
/tmp/herbein1/spack-stage/spack-stage-flux-core-0.23.0-2ac2sbdm7aolnplhwir3b2rizxk2c7px/spack-src/src/modules/job-manager/jobtap.c: In function 'jobtap_load':
/tmp/herbein1/spack-stage/spack-stage-flux-core-0.23.0-2ac2sbdm7aolnplhwir3b2rizxk2c7px/spack-src/src/modules/job-manager/jobtap.c:417:9: error: 'strncpy' specified bound 128 equals destination size [-Werror=stringop-truncation]
  417 |         strncpy (errp->text, flux_plugin_strerror (p), sizeof (errp->text));
      |         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[3]: *** [Makefile:1095: jobtap.lo] Error 1
make[3]: *** Waiting for unfinished jobs....
make[3]: Leaving directory '/tmp/herbein1/spack-stage/spack-stage-flux-core-0.23.0-2ac2sbdm7aolnplhwir3b2rizxk2c7px/spack-src/spack-build/src/modules/job-manager'
make[2]: *** [Makefile:497: all-recursive] Error 1
make[2]: Leaving directory '/tmp/herbein1/spack-stage/spack-stage-flux-core-0.23.0-2ac2sbdm7aolnplhwir3b2rizxk2c7px/spack-src/spack-build/src/modules'
make[1]: *** [Makefile:507: all-recursive] Error 1
make[1]: Leaving directory '/tmp/herbein1/spack-stage/spack-stage-flux-core-0.23.0-2ac2sbdm7aolnplhwir3b2rizxk2c7px/spack-src/spack-build/src'
make: *** [Makefile:580: all-recursive] Error 1
```

--- a/src/modules/job-manager/jobtap.c
+++ b/src/modules/job-manager/jobtap.c
@@ -421,7 +421,9 @@ done:
     return 0;
 error:
     if (errp && errp->text[0] == '\0')
-        strncpy (errp->text, flux_plugin_strerror (p), sizeof (errp->text));
+        strncpy (errp->text,
+                 flux_plugin_strerror (p),
+                 sizeof (errp->text) - 1);
     flux_plugin_destroy (p);
     return -1;
 }

